mail_tracking/static/src/js/failed_message/discuss.js

```diff
@@ -0,0 +1,398 @@
+        },
+
+        /**
+         * Overrides to update 'set all as reviewed' button
+         *
+         * @Override
+         */
+        _updateControlPanelButtons: function (thread) {
+            // Set All Reviewed
+            if (thread.getID() === 'mailbox_failed') {
+                this.$buttons
+                    .find('.o_mail_discuss_button_set_all_reviewed')
+                    .removeClass('d-none d-md-inline-block')
+                    .addClass('d-none d-md-inline-block');
+            } else {
+                this.$buttons
+                    .find('.o_mail_discuss_button_set_all_reviewed')
+                    .removeClass('d-none d-md-inline-block')
+                    .addClass('d-none');
+            }
+
+            this._super.apply(this, arguments);
+        },
+
+        /**
+         * Overrides to update 'set all as reviewed' button
+         *
+         * @Override
+         */
+        _updateButtonStatus: function (disabled, type) {
+            if (this._thread.getID() === 'mailbox_failed') {
+                this.$buttons
+                    .find('.o_mail_discuss_button_set_all_reviewed')
+                    .toggleClass('disabled', disabled);
+                //  Display Rainbowman when all inbox messages are reviewed
+                // through 'TOGGLE TRACKING STATUS' or marking last failed
+                // message as reviewed
+                if (disabled && type === 'toggle_tracking_status') {
+                    this.trigger_up('show_effect', {
+                        message: _t(
+                            "Congratulations, your failed mailbox is empty"),
+                        type: 'rainbow_man',
+                    });
+                }
+            }
+        },
+
+        /**
+         * Overrides to update messages in 'failed' mailbox thread
+         *
+         * @Override
+         */
+        _onMessageUpdated: function (message, type) {
+            var self = this;
+            var currentThreadID = this._thread.getID();
+            if (currentThreadID === 'mailbox_failed' && !message.isFailed()) {
+                this._thread.fetchMessages(this.domain)
+                    .then(function () {
+                        var options = self._getThreadRenderingOptions();
+                        self._threadWidget.removeMessageAndRender(
+                            message.getID(), self._thread, options)
+                            .then(function () {
+                                self._updateButtonStatus(
+                                    !self._thread.hasMessages(), type);
+                            });
+                    });
+            } else {
+                //  Workaround to avoid call '_fetchAndRenderThread' and refetch
+                // thread messages.
+                this._super.apply(this, arguments);
+            }
+        },
+
+        /**
+        *  Overrides to hide reply in the 'failed' mailbox, as this feature does
```

Comment:
Good docs are both specific and short:

```suggestion
        * Hide reply feature in the 'failed' mailbox, where it has no sense.
        *
        * Show instead 'Retry' and 'Set as reviewed' buttons.
```